File: ProjectStarbucks/Break-Continue-Nodes/Node.java
/******************************************************************************

Author: Dominic "Seth" Jones-Jackson (F)
Date: October 5, 2020

*******************************************************************************/
public class Node
{
    int value;
    Node link;
    
    public Node(int input)
    {
        value = input;
        link = null;
    }
    // End of Constructor;
    
    public void Add(int input)
    {
        if(link == null)
            link = new Node(input);
        else
            link.Add(input);
    }
    // End of Add method
    
    public int Count(int iter)
    {
        iter++;
        if(link != null)
            iter = link.Count(iter);
        
        return iter;
    }
    // End of Count method
    
    public static int[] ToArray(Node input)
    {
        int length = input.Count(0);
        int i = 0;
        int[] output = new int[length];
        
        while (input.link != null)
        {
            output[i] = input.value;
            i++;
            input = input.link;
        }
        // End of While loop
        output[i] = input.value;
        
        return output;
    }
    // End of ToArray method
}
// End of Node class 

File: ProjectMakeItSo/HashTable/HashTable/Node.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace HashTable
{
    class Node
    {
        private Node next;
        private Contact value;

        public Contact Value { get => value; set => this.value = value; }
        public Node Next { get => next; set => next = value; }

        public Node(Contact value)
        {
            this.next = null;
            this.value = value;
        }
        // End of Constructor

        public void Add(Contact value)
        {
            Node input = new Node(value);
            if (this.next == null)
                this.next = input;
            else
                this.next.Add(value);
        }
        // End of Add method

        public int Count(int output = 0)
        {
            output++;
            if (this.next != null)
                output = this.next.Count(output);

            return output;
        }
        // End of Count method

        public static Contact[] Array(Node input)
        {
            Contact[] output = new Contact[input.Count()];

            for(int i = 0; i < output.Length; i++)
            {
                output[i] = input.Value;
                input = input.Next;
            }
            // End of For loop

            return output;
        }
        // End of static Array method
    }
    // End of Node class
}
// End of namespace